eclipse-collections/src/main/java/org/eclipse/collections/impl/list/fixed/FixedSizeListFactoryImpl.java

```diff
@@ -10,10 +10,12 @@
 
 package org.eclipse.collections.impl.list.fixed;
 
+import com.google.auto.service.AutoService;
 import org.eclipse.collections.api.factory.list.FixedSizeListFactory;
 import org.eclipse.collections.api.list.FixedSizeList;
 import org.eclipse.collections.impl.utility.Iterate;
 
+@AutoService(FixedSizeListFactory.class)
 public enum FixedSizeListFactoryImpl implements FixedSizeListFactory
 {
     INSTANCE;
```

Comment:
This actually doesn't work yet because I left these Impl classes as singleton enums. I need to convert them to plain old classes with a zero-arg constructor so that they can be instantiated through reflection. this is a fairly breaking change so I wanted to highlight it.

Reply:
Why is that a breaking change?

Reply:
I guess I can create a public INSTANCE constant to keep it source compatible.

Reply:
Not that it matters, but what breaks binary compatibility? Am I misunderstanding that all you're doing is removing the private constructor?

Reply:
I am not 100% sure that it's a binary compatibility break. But I know that using the enum keyword changes the bytecode in a few ways. Now that I'm thinking about it, I know a few more things I can do to improve compatibility.